### PR TITLE
reverse proxy nixos module for nginx

### PR DESCRIPTION
**Pull request by ts468**
Follow-up PR on #9103

### CODE REVIEW

**Comment by ts468** [comment]
@domenkozar Here we go ;)

**Comment by domenkozar** [comment]
Thanks!

**Comment by jagajaga** [comment]
Please don't forget to properly name commit. And fix conflicts with master.

**Comment by wmertens** [comment]
:+1:

File: nixos/modules/services/web-servers/nginx/reverse_proxy.nix [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+      reverseProxies = mkOption {
+        type = types.attrsOf (types.submodule (
+          {
+            options = {
+              proxy = mkOption {
+                type = types.str;
+                default = [];
+                description = ''
+                  Exclude files and directories matching these patterns.
```
Comment (trishume) [inline nixos/modules/services/web-servers/nginx/reverse_proxy.nix]: Don't think this description is correct. Also copy-pasted 2 times below.

File: nixos/modules/services/web-servers/nginx/reverse_proxy.nix [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    services.nginx.config = mkBefore ''
+      worker_processes 1;
+      error_log logs/error.log debug;
+      pid logs/nginx.pid;
+      events {
+         worker_connections  1024;
+      }
+    '';
+
+    services.nginx.httpConfig = mkBefore ''
```
Comment (trishume) [inline nixos/modules/services/web-servers/nginx/reverse_proxy.nix]: Great to have these sane defaults. These should maybe even get moved to the main nginx module eventually, they can probably stay here for this PR though.

**Comment by trishume** [comment]
This looks quite useful. If @ts468 doesn't fix this up in the next week or two, I can fix this up for them and maybe in another PR also add an option for serving static files from a directory if available ([example of a config that could use this here](https://github.com/trishume/nixfiles/blob/master/rate-with-science.nix#L87)).

**Comment by ts468** [comment]
@trishume Thanks for your interest in this PR. The PR isn't dead, yet ;). It's just that @domenkozar and me still need to figure out how and where to include the new options. It's not clear yet whether to put them into a separate branch, a separate module or into the nginx module directly. Just keep pinging us to remind us to make some progress on it ;)

**Comment by globin** [comment]
I'm not sure if this is the way to go, as this may be blocking further work on abstracting away the nginx config similar to apache config or puppet (or similar) modules for nginx, by possibly making it necessary to make backward incompatible changes?

**Comment by ts468** [comment]
@globin Thanks, I just think that once the abstraction over nginx and apache is really in place we just make the necessary, backward incompatible changes. I'm not really afraid of that.

**Comment by fpletz** [comment]
Here is our proposal for the nginx module: #15862

**Comment by joachifm** [comment]
Looks like this has been superceded. Please reopen if this is incorrect.

**Pull request closed without merging**